### PR TITLE
Fix(translation) default locale file not updated

### PR DESCRIPTION
**Pull request by p10ns11y**
Addressing https://github.com/mxstbr/react-boilerplate/issues/1363

- [x] You have followed our [**contributing guidelines**](https://github.com/mxstbr/react-boilerplate/blob/master/.github/CONTRIBUTING.md)
- [x] double-check your branch is based on `dev` and targets `dev` 
- [x] Pull request has tests (we are going for 100% coverage!)
- [x] Code is well-commented, linted and follows project conventions
- [x] Description explains the issue/use-case resolved and auto-closes related issues

### CODE REVIEW

**Comment by mxstbr** [comment]
Note: this breaks `npm run generate language` and possibly `npm run clean`.

I'm not a fan of moving locale related stuff to the `App` container. `i18n.js` exists so people who just want a running and working i18n setup don't have to worry about it and just have an extra file. Everybody using react-boilerplate will touch the `App` container eventually, so as little things as possible should be there.

In general I love where you're going, but can we do it while leaving the i18n stuff contained in `i18n.js` please?

**Comment by p10ns11y** [comment]
@mxstbr : Done!

There is possibility that `npm run generate language` just add the locale to `appLocales`
then we can do the rest of things instead of pushing the code template. That I will do as separate PR later sometime

File: internals/scripts/extract-intl.js [hunk 1]
```diff
@@ -14,14 +14,14 @@ const pkg = require('../../package.json');
 const presets = pkg.babel.presets;
 const plugins = pkg.babel.plugins || [];
 
-const i18n = require('../../app/i18n');
-import { DEFAULT_LOCALE } from '../../app/containers/App/constants';
-
 require('shelljs/global');
 
+import { DEFAULT_LOCALE } from '../../app/containers/App/constants';
+import { appLocales } from '../../app/i18n';
```
Comment (mxstbr) [inline internals/scripts/extract-intl.js]: I don't think `import`s work here?! This is ran in Node, untranspiled, which doesn't have `import`?

Comment (p10ns11y) [inline internals/scripts/extract-intl.js]: @mxstbr : `import { DEFAULT_LOCALE } from '../../app/containers/App/constants';` was already there in master branch hence I just followed.  

`npm run extract-intl` => ` babel-node --presets latest,stage-0 -- ./internals/scripts/extract-intl.js` 

Hence `import` very well works.  

However `node internals/scripts/extract-intl.js` will not work as you said. 
https://github.com/p10ns11y/react-boilerplate/blob/d109f617f57eaae5fc7a08f7c91ad40477ab29be/internals/scripts/extract-intl.js#L100

Using arrow `async` function which is not available in node as well. Then we need to rewrite the entire script native node.

**Comment by p10ns11y** [comment]
@gihrig Do I miss anything ?

**Comment by gihrig** [comment]
From #1363 

> Translation file is not updated: just update any defaultMessages in some container's messages

Run `npm run extract-intl` Works as expected ✨ 

> Delete-proof: delete translations folder and run npm run extract-intl

image

**Comment by p10ns11y** [comment]
@gihrig : Well, the very first commit of this pull request solved that https://github.com/mxstbr/react-boilerplate/pull/1364/commits/8902bfc605600bc05de177293810ca6740c21258

By moving  `appLocales` to `app/containers/App/constants` I was able to get rid of `i18n` dependency as `i18n` importing translations files.

Later @mxstbr suggested to keep the `i18n` and `app` as it. Then I limited the issue and pull request solving only the problem of extraction failure upon new updates. 

=> `npm generate language` should add locales in different file that has to imported in `i18n`

Would you suggest of rewriting the `npm generate language` robust way that only requires the values of `appLocales` then we can iterate throw and dynamically import.

**Comment by gihrig** [comment]
@p10ns11y 

>Would you suggest of rewriting the npm generate language robust way that only requires the values of appLocales then we can iterate throw and dynamically import.

Honestly I'm not the best person to ask about that as I don't have much interest in the i18n feature as architected. But if we have it should be reliable, IMO.

I'm not sure that recovery from a deleted `app/translations` folder is necessary, but the system probably should recover from missing `xx.json` files, since the purpose of `extract-intl` is to modify `xx.json` files. The error that is presented now, as above is ugly, but meaningful. So maybe it's OK. As I said, I'm not the best one to ask...

**Pull request closed without merging**